enhancements/autoscaling/provisioning-request-enablement.md

```diff
@@ -0,0 +1,337 @@
+
+1. Unauthorized creation of infrastructure resources. Because the ProvisioningRequest
+  resource will lead to infrastructure resource creation, it must have a proper role and
+  permissions for users to create. This will follow the same role-based authentication
+  control pattern as other Kubernetes resources, but is worth noting as creation of
+  ProvisioningRequests could lead to the creation of infrastructure resources.
+2. Confusion for users around the use of ProvisioningRequests. The Cluster Autoscaler will
+  only react to ProvisioningRequests that are created in the `openshift-machine-api`
+  namespace. This will be clearly illustrated with examples in the OpenShift product
+  documentation.
+
+### Drawbacks
+
+One drawback to this approach is the inclusion of ProvisioningRequest as a supported type
+on core OpenShift. This adds complexity to OpenShift not only in operation but in support
+as well. It is entirely possible that users could install the ProvisioningRequest
+custom resource definition on their own.
+
+The trade-offs for having the user install the ProvisioningRequest definition on their own
+come in two areas: user experience, and Cluster Autoscaler configuration.
+
+The user experience for managing custom resource definitions that interact with Kubernetes
+components, such as the Cluster Autoscaler, requires users to understand the version
+changes and differences. Users need to understand when the project versions have changed
+to ensure that they are installing the correct versions of an API resource. Further, they
+will need to manually migrate on upagrades of the underlying platform.
+
+In addition to installing the ProvisioningRequest, users will need to configure the Cluster
+Autoscaler to reconcile the resources as well. This will require the addition of a new
+field in the ClusterAutoscaler resource on OpenShift.
+
+## Open Questions
+
+1. Do we want to create a field in the ClusterAutoscaler resource to allow disabling of the
+  ProvisioningRequest?
+
+## Test Plan
+
+* Unit tests in openshift/cluster-autoscaler-operator to confirm deployment configuration.
+* End-to-end tests openshift/cluster-api-actuator-pkg to confirm the ProvisioningRequest
+  behavior with Cluster Autoscaler.
+* Test upgrade to confirm that feature is available.
+* Test downgrade to confirm that feature is disabled.
+
+## Graduation Criteria
+
+This feature is currently released as stable in the Kubernetes community. As such, its
+graduation within OpenShift will be accelerated. This feature will be created behind
+the `TechPreviewNoUpgrade` feature gate. The feature gate will be removed once testing
+has confirmed that the feature does not introduce regressions, and after the feature
+has been made available for internal feedback following review.
+
+### Dev Preview -> Tech Preview
+
+- Feature gate for API changes and Cluster Autoscaler behavior
+
+### Tech Preview -> GA
+
+- Unit testing passing
```

Comment:
How would you feel about colloborating on a Kueue + ProvRequest in the next release?

ie 4.19 we TP this with the API enabled.
in 4.20, we can define an OEP here or in Kueue to walk through how one should enable this.

There has been a lot of questions on how this should work with Kueue. And given that Kueue is a major user of this, I think it would make sense to define what we want in either Kueue or ProvRequest.

Reply:
i think we can bring the ProvReq crd to ocp without solving all the kueue integrations. i see this as cluster autoscaler providing an api, we could certainly coordinate around how the api is exposed.

Reply:
there is a line about passing kueue integration in there